scipy/optimize/tests/test_optimize.py

```diff
@@ -2180,6 +2181,33 @@ def func(z, *params):
                           match=r'Either final optimization did not succeed'):
             optimize.brute(func, self.rranges, args=self.params, disp=True)
 
+        message = "Polishing is incompatible with integrality constraints"
+        with pytest.warns(OptimizeWarning, match=message):
+            optimize.brute(func, self.rranges, args=self.params,
+                           integrality=True)
+
+    def test_integrality(self):
+        # test fitting discrete distribution to data
+        rng = np.random.default_rng(6519843218105)
+        dist = stats.nbinom
+        shapes = (5, 0.5)
+        x = dist.rvs(*shapes, size=10000, random_state=rng)
+
+        def func(free_params, *args):
+            dist, x = args
+            # negative log-likelihood function
+            ll = -np.log(dist.pmf(x, *free_params)).sum(axis=-1)
+            if np.isnan(ll):  # occurs when x is outside of support
+                ll = np.inf   # we don't want that
+            return ll
+
+        integrality = [True, False]
+        bounds = [(1, 18), (0, 0.95)]
+
+        res = optimize.brute(func, bounds, args=(dist, x), finish=False,
+                             integrality=integrality, Ns=20)
+        assert_allclose(res, shapes, atol=1e-14)
```

Comment:
It should be noted here that the second fit parameter (lying in the range `(0, 0.95)`) is still only going to take one of a set of discrete values because of the way that `brute` works. e.g. if you set `Ns=21` then this assertion will fail.

Reply:
Yes, I know. `brute` is not very good on its own, but this test shows that treating the integer-constrained variable and the continuous variable differently and as expected.

Reply:
Ok. If that's the case then add a comment to the test so that this is understood in the future.

Reply:
OK.

Reply:
Done.